mava/environment_loop.py

```diff
@@ -243,31 +243,41 @@ def should_run_loop(eval_interval_condition: Tuple) -> bool:
             evaluation_duration = eval_duration_condition[1]
 
         while True:
-            if (not environment_loop_schedule) or (
-                should_run_loop(eval_interval_condition)
-            ):
-                if environment_loop_schedule:
-                    results = self.run_episode()
-                    # Get first result dictionary
-                    for _ in range(evaluation_duration - 1):
-                        # Add consecutive evaluation run data
+            try:
+                if (not environment_loop_schedule) or (
```

Comment:
Can we maybe create a step function for all the code in the try? Then we just try and catch over the step function?

Reply:
Nice idea I will do that!

Reply:
Thanks :slightly_smiling_face: